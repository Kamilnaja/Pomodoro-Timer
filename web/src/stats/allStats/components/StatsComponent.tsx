import { Button, ButtonGroup, Table } from 'react-bootstrap';
import { getCurrentMonth, getCurrentYear } from '../../../shared/scripts/utils';
import './stats.component.scss';
import { StatsComponentProps } from './StatsComponentProps';

let pageMonth = getCurrentMonth();
let pageYear = getCurrentYear();

export const StatsComponent = (props: StatsComponentProps) => {
  const getPreviousMonth = () => {
    if (pageMonth === 0) {
      pageMonth = 11;
      pageYear = pageYear - 1;
    } else {
      pageMonth = pageMonth - 1;
    }

    props.handleGetStats(pageYear, pageMonth);
  };

  const getNextMonth = (): void => {
    if (pageMonth === 11) {
      pageMonth = 0;
      pageYear = pageYear + 1;
    } else {
      pageMonth = pageMonth + 1;
    }

    props.handleGetStats(pageYear, pageMonth);
  };

  // const renderLineChart = (
  //   <LineChart width={600} height={300} data={props.stats}>
  //     <Line type="monotone" dataKey="count" stroke="#8884d8" />
  //   </LineChart>
  // );

  return (
    <div className="stats">
      {/* {renderLineChart} */}
      <ButtonGroup className="stats__navigation navigation">
        <Button
          className="navigation__button"
          onClick={() => getPreviousMonth()}
          disabled={!props.stats.hasPreviousPeriod}
        >
          &lt; prev
        </Button>
        <Button className="navigation__button" onClick={() => getNextMonth()} disabled={!props.stats.hasNextPeriod}>
          next &gt;
        </Button>
      </ButtonGroup>
      <Table className="stats__table table">
        <caption>
          Results from: {pageMonth + 1}.{pageYear}
        </caption>
        <thead className="table__head">
          <tr>
            <th>Date</th>
            <th>Number of pomodoros</th>
          </tr>
        </thead>
        <tbody>
          {props.stats.pomodoros?.map((v, idx) => (
            <tr key={idx}>
              <td className="table__date">{v.date}</td>
              <td className="table__count">{v.count}</td>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );
};
