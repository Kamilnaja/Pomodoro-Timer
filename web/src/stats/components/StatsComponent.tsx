import { getCurrentMonth, getCurrentYear } from '../../shared/scripts/utils';
import { StatsComponentProps } from '../store/interfaces/statsInterfaces';
import './stats.component.scss';

let pageMonth = getCurrentMonth();
let pageYear = getCurrentYear();

export const StatsComponent = (results: StatsComponentProps) => {
  const lastResult = results.stats[0];
  const isAnyPomodoroDoneToday = (): boolean => {
    // get last stats days and reverse to allow date comparition
    const reverseLastStatsDate: string = lastResult?.date.split('-').reverse().join('-');
    return new Date(reverseLastStatsDate).toDateString() === new Date().toDateString();
  };

  const getPreviousMonth = () => {
    if (pageMonth === 0) {
      pageMonth = 11;
      pageYear = pageYear - 1;
    } else {
      pageMonth = pageMonth - 1;
    }

    results.handleGetStats(pageYear, pageMonth);
  };

  const getNextMonth = (): void => {
    if (pageMonth === 11) {
      pageMonth = 0;
      pageYear = pageYear + 1;
    } else {
      pageMonth = pageMonth + 1;
    }

    results.handleGetStats(pageYear, pageMonth);
  };

  const shouldShowNextMonth = (): boolean => {
    return getCurrentYear() <= pageYear && getCurrentMonth() <= pageMonth;
  };

  return (
    <div className="stats">
      <h2 className="stats__today-result">
        Today you have made: {isAnyPomodoroDoneToday() ? lastResult?.count : 0} pomodoros
      </h2>
      <div className="stats__navigation navigation">
        <button className="navigation__arrow" onClick={() => getPreviousMonth()}>
          &lt; prev
        </button>
        <button className="navigation__arrow" onClick={() => getNextMonth()} disabled={shouldShowNextMonth()}>
          next &gt;
        </button>
      </div>
      <table className="stats__table table">
        <caption>
          Results from: {pageMonth + 1}.{pageYear}
        </caption>
        <thead className="table__head">
          <tr>
            <th>Date</th>
            <th>Number of pomodoros</th>
          </tr>
        </thead>
        <tbody>
          {results.stats?.map((v, idx) => (
            <tr key={idx}>
              <td className="table__date">{v.date}</td>
              <td className="table__count">{v.count}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};
