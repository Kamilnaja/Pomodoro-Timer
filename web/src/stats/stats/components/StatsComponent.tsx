import { Button, ButtonGroup, Container } from 'react-bootstrap';
import { getCurrentMonth, getCurrentYear } from '../../../shared/scripts/utils';
import StatsTableContainer from '../../statsTable/container/StatsTableContainer';
import './stats.component.scss';
import { StatsComponentProps } from './StatsComponentProps';

let pageMonth = getCurrentMonth();
let pageYear = getCurrentYear();

export const StatsComponent = (props: StatsComponentProps) => {
  const handleChangeMonth = (isNext: boolean) => {
    const nextOrPreviousMonth = isNext ? 1 : -1;
    const dt = new Date(pageYear, pageMonth);
    dt.setMonth(dt.getMonth() + nextOrPreviousMonth);

    pageMonth = dt.getMonth();
    pageYear = dt.getFullYear();

    props.handleGetStats(dt.getFullYear(), dt.getMonth());
  };

  return (
    <Container>
      <div className="stats">
        {/* {renderLineChart} */}
        <ButtonGroup className="stats__navigation navigation">
          <Button
            className="navigation__button navigation__button--prev"
            onClick={() => handleChangeMonth(false)}
            disabled={!props.stats.hasPreviousPeriod}
          >
            &lt; prev
          </Button>
          <Button
            className="navigation__button navigation__button--next"
            onClick={() => handleChangeMonth(true)}
            disabled={!props.stats.hasNextPeriod}
          >
            next &gt;
          </Button>
        </ButtonGroup>

        <StatsTableContainer
          pageMonth={pageMonth}
          pageYear={pageYear}
          pomodoros={props.stats.pomodoros}
          handleGetSettings={props.handleGetSettings}
          handleSaveSettings={props.handleSaveSettings}
          settingsState={props.settings}
          toggleDisplayDirection={props.toggleDisplayDirection}
          toggleDisplayEmptyDays={props.toggleDisplayEmptyDays}
        ></StatsTableContainer>
      </div>
    </Container>
  );
};
