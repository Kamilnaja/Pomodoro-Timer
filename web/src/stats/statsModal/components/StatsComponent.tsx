import { Button, ButtonGroup, Container } from 'react-bootstrap';
import { StatsTableContainer } from '../../statsTable/container/StatsTableContainer';
import './stats.component.scss';
import { StatsComponentProps } from './StatsComponentProps';

export const StatsComponent = (props: StatsComponentProps) => {
  const changeMonth = (increment: boolean) => {
    increment ? props.incrementMonth() : props.decrementMonth();
    props.handleGetStats(props.date.getFullYear(), props.date.getMonth());
  };

  return (
    <Container>
      <div className="stats">
        <ButtonGroup className="stats__navigation navigation">
          <Button
            className="navigation__button navigation__button--prev"
            onClick={() => changeMonth(false)}
            disabled={!props.stats.hasPreviousPeriod}
          >
            &lt; prev
          </Button>
          <Button
            className="navigation__button navigation__button--next"
            onClick={() => changeMonth(true)}
            disabled={!props.stats.hasNextPeriod}
          >
            next &gt;
          </Button>
        </ButtonGroup>

        <StatsTableContainer
          pageMonth={props.date.getMonth()}
          pageYear={props.date.getFullYear()}
          pomodoros={props.stats.pomodoros}
          handleGetSettings={props.handleGetSettings}
          handleSaveSettings={props.handleSaveSettings}
          settingsState={props.settings}
          toggleDisplayDirection={props.toggleDisplayDirection}
          toggleDisplayEmptyDays={props.toggleDisplayEmptyDays}
        ></StatsTableContainer>
      </div>
    </Container>
  );
};
