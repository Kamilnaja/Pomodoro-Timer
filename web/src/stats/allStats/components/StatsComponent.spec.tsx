import { shallow } from 'enzyme';
import { createStats } from '../../testing/stats.test.data';
import { StatsComponent } from './StatsComponent';
import { shallowToJson } from 'enzyme-to-json';

describe('StatsComponent', () => {
  let statsComponent = shallow(<StatsComponent stats={createStats()} handleGetStats={() => {}} />);

  it('Should match snapshot', () => {
    expect(statsComponent).not.toBeNull();
    expect(shallowToJson(statsComponent)).toMatchSnapshot();
  });
});
