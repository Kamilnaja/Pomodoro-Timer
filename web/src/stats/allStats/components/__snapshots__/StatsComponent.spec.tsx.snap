// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsComponent Should match snapshot 1`] = `
<div
  className="stats"
>
  <div
    className="stats__navigation navigation"
  >
    <button
      className="navigation__button"
      onClick={[Function]}
    >
      &lt; prev
    </button>
    <button
      className="navigation__button"
      disabled={true}
      onClick={[Function]}
    >
      next &gt;
    </button>
  </div>
  <ForwardRef
    className="stats__table table"
  >
    <caption>
      Results from: 
      2
      .
      2021
    </caption>
    <thead
      className="table__head"
    >
      <tr>
        <th>
          Date
        </th>
        <th>
          Number of pomodoros
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <td
          className="table__date"
        >
          21-02-2021
        </td>
        <td
          className="table__count"
        >
          8
        </td>
      </tr>
      <tr
        key="1"
      >
        <td
          className="table__date"
        >
          20-02-2021
        </td>
        <td
          className="table__count"
        >
          15
        </td>
      </tr>
      <tr
        key="2"
      >
        <td
          className="table__date"
        >
          19-02-2021
        </td>
        <td
          className="table__count"
        >
          16
        </td>
      </tr>
      <tr
        key="3"
      >
        <td
          className="table__date"
        >
          18-02-2021
        </td>
        <td
          className="table__count"
        >
          19
        </td>
      </tr>
      <tr
        key="4"
      >
        <td
          className="table__date"
        >
          17-02-2021
        </td>
        <td
          className="table__count"
        >
          18
        </td>
      </tr>
      <tr
        key="5"
      >
        <td
          className="table__date"
        >
          16-02-2021
        </td>
        <td
          className="table__count"
        >
          11
        </td>
      </tr>
      <tr
        key="6"
      >
        <td
          className="table__date"
        >
          15-02-2021
        </td>
        <td
          className="table__count"
        >
          16
        </td>
      </tr>
      <tr
        key="7"
      >
        <td
          className="table__date"
        >
          14-02-2021
        </td>
        <td
          className="table__count"
        >
          13
        </td>
      </tr>
      <tr
        key="8"
      >
        <td
          className="table__date"
        >
          13-02-2021
        </td>
        <td
          className="table__count"
        >
          14
        </td>
      </tr>
      <tr
        key="9"
      >
        <td
          className="table__date"
        >
          12-02-2021
        </td>
        <td
          className="table__count"
        >
          9
        </td>
      </tr>
      <tr
        key="10"
      >
        <td
          className="table__date"
        >
          11-02-2021
        </td>
        <td
          className="table__count"
        >
          11
        </td>
      </tr>
      <tr
        key="11"
      >
        <td
          className="table__date"
        >
          10-02-2021
        </td>
        <td
          className="table__count"
        >
          13
        </td>
      </tr>
      <tr
        key="12"
      >
        <td
          className="table__date"
        >
          09-02-2021
        </td>
        <td
          className="table__count"
        >
          13
        </td>
      </tr>
      <tr
        key="13"
      >
        <td
          className="table__date"
        >
          08-02-2021
        </td>
        <td
          className="table__count"
        >
          18
        </td>
      </tr>
      <tr
        key="14"
      >
        <td
          className="table__date"
        >
          07-02-2021
        </td>
        <td
          className="table__count"
        >
          14
        </td>
      </tr>
      <tr
        key="15"
      >
        <td
          className="table__date"
        >
          06-02-2021
        </td>
        <td
          className="table__count"
        >
          16
        </td>
      </tr>
      <tr
        key="16"
      >
        <td
          className="table__date"
        >
          05-02-2021
        </td>
        <td
          className="table__count"
        >
          13
        </td>
      </tr>
      <tr
        key="17"
      >
        <td
          className="table__date"
        >
          04-02-2021
        </td>
        <td
          className="table__count"
        >
          12
        </td>
      </tr>
      <tr
        key="18"
      >
        <td
          className="table__date"
        >
          03-02-2021
        </td>
        <td
          className="table__count"
        >
          16
        </td>
      </tr>
      <tr
        key="19"
      >
        <td
          className="table__date"
        >
          02-02-2021
        </td>
        <td
          className="table__count"
        >
          15
        </td>
      </tr>
      <tr
        key="20"
      >
        <td
          className="table__date"
        >
          01-02-2021
        </td>
        <td
          className="table__count"
        >
          17
        </td>
      </tr>
    </tbody>
  </ForwardRef>
</div>
`;
